src/arrayMethodJoin.js

```diff
@@ -5,7 +5,47 @@
  */
 function applyCustomJoin() {
   [].__proto__.join2 = function(separator) {
-    // write code here
+    let string = '';
+    let currentSeparator = separator;
+
+    if (this.length === 0) {
+      return string;
+    }
+
+    if (this.length === 1 && (this[0] === undefined || this[0] === null)) {
+      return string;
+    }
+
+    if (this.length === 1) {
+      string = `${this[0]}`;
+
+      return string;
+    }
+
+    if (currentSeparator === undefined) {
+      currentSeparator = ',';
+    }
```

Comment:
I think it's better to write it like this

Reply:
function(separator = ',')

Reply:
Thanks, you are right)